addons/purchase_stock/models/product.py

```diff
@@ -36,36 +36,40 @@ class ProductProduct(models.Model):
 
     purchase_order_line_ids = fields.One2many('purchase.order.line', 'product_id', string="PO Lines")  # used to compute quantities
     monthly_demand = fields.Float(compute='_compute_monthly_demand')
-    suggested_qty = fields.Integer(compute="_compute_suggested_quantity")
+    suggested_qty = fields.Integer(compute='_compute_suggested_quantity', search='_search_product_with_suggested_quantity')
     suggest_estimated_price = fields.Float(compute='_compute_suggest_estimated_price')
 
     @api.depends("monthly_demand")
     @api.depends_context("suggest_based_on", "suggest_days", "suggest_percent", "warehouse_id")
     def _compute_suggested_quantity(self):
-        """ IMPROVE: computes too many time for one suggestion """
         ctx = self.env.context
-        for product in self:
-            if not ctx.get("suggest_based_on"):
-                product.suggested_qty = 0
-                continue
-            elif ctx.get("suggest_based_on") == "actual_demand":
+        self.suggested_qty = 0
+        if not ctx.get("suggest_based_on"):
+            return
+        elif ctx.get("suggest_based_on") == "actual_demand":
```

Comment:
Just a suggestion, keep your prefered version
```suggestion
        if ctx.get("suggest_based_on") == "actual_demand":
```
1/2